ffi/src/sec_winnt_auth_identity.rs

```diff
@@ -263,7 +265,7 @@ pub unsafe fn auth_data_to_identity_buffers_w(
 
         // only marshaled smart card creds starts with '@' char
         #[cfg(feature = "scard")]
-        if user[0] == b'@' {
+        if user[0] == b'@' && CredIsMarshaledCredentialW(user.as_ptr() as *const _) != 0 {
```

Comment:
calling CredIsMarshaledCredentialW should be enough, it checks for @@ under the hood anyway. maybe the only potential issue is that CredIsMarshaledCredentialW could return true for types other than 1

Reply:
done